node/runtime/src/impls.rs

```diff
@@ -155,50 +144,67 @@ mod tests {
 		// Current saturation in terms of weight
 		let s = block_weight;
 
-		let fm = (v * (s/m - ss/m)) + (v.powi(2) * (s/m - ss/m).powi(2)) / 2.0;
-		let addition_fm = Fixed64::from_parts((fm * 1_000_000_000_f32) as i64);
+		let fm = v * (s/m - ss/m) + v.powi(2) * (s/m - ss/m).powi(2) / 2.0;
+		let addition_fm = Fixed64::from_parts((fm * 1_000_000_000_f32).round() as i64);
 		previous.saturating_add(addition_fm)
 	}
 
-	fn fm(parts: i64) -> Fixed64 {
+	fn feemul(parts: i64) -> Fixed64 {
 		Fixed64::from_parts(parts)
 	}
 
+	fn run_with_system_weight<F>(w: Weight, assertions: F) where F: Fn() -> () {
+		let mut t: runtime_io::TestExternalities =
+			system::GenesisConfig::default().build_storage::<Runtime>().unwrap().into();
+		t.execute_with(|| {
+			System::set_block_limits(w, 0);
+			assertions()
+		});
+	}
+
 	#[test]
 	fn fee_multiplier_update_poc_works() {
 		let fm = Fixed64::from_rational(0, 1);
 		let test_set = vec![
-			// TODO: this has a rounding error and fails.
-			// (0, fm.clone()),
+			(0, fm.clone()),
 			(100, fm.clone()),
 			(target(), fm.clone()),
 			(max() / 2, fm.clone()),
 			(max(), fm.clone()),
 		];
 		test_set.into_iter().for_each(|(w, fm)| {
-			assert_eq!(
-				fee_multiplier_update(w, fm),
-				FeeMultiplierUpdateHandler::convert((w, fm)),
-				"failed for weight {} and prev fm {:?}",
-				w,
-				fm,
-			);
+			run_with_system_weight(w, || {
+				assert_eq_error_rate!(
+					fee_multiplier_update(w, fm).into_inner(),
+					TargetedFeeAdjustment::convert(fm).into_inner(),
+					5,
+				);
+			})
 		})
 	}
 
 	#[test]
 	fn empty_chain_simulation() {
 		// just a few txs per_block.
-		let block_weight = 1000;
-		let mut fm = Fixed64::default();
-		let mut iterations: u64 = 0;
-		loop {
-			let next = FeeMultiplierUpdateHandler::convert((block_weight, fm));
-			fm = next;
-			if fm == Fixed64::from_rational(-1, 1) { break; }
-			iterations += 1;
-		}
-		println!("iteration {}, new fm = {:?}. Weight fee is now zero", iterations, fm);
+		let block_weight = 0;
```

Comment:
is it on purpose that block weight changed from 1000 to 0 ?

Reply:
yeah just makes a bit more sense: the test is supposed to simulate an _empty_ chain. Nonetheless the test is not that important as you see and just runs a simulation.